pkg/util/util.go

```diff
@@ -857,15 +863,40 @@ func Unzip(src, dest string) ([]string, error) {
 	defer r.Close()
 
 	for _, f := range r.File {
-
 		// Store filename/path for returning and using later on
 		index := strings.Index(f.Name, "/")
 		filename := f.Name[index+1:]
 		if filename == "" {
 			continue
 		}
+
+		// if sparseCheckoutDir has a pattern
+		match, err := filepath.Match(pathToUnzip, filename)
```

Comment:
I think we need to be careful here with pattern matching:

Using the [quarkus devfile](https://raw.githubusercontent.com/eclipse/che-devfile-registry/8cf46dc0e71c3ff6a9631965bed74c9202b13d07/devfiles/quarkus/devfile.yaml), we're pattern matching all of the folders that are prefixed with `getting-started`, rather than just the one folder `getting-started/`:
```
PS D:\ododev\sparse> odo create --downloadSource                                                                        Experimental mode is enabled, use at your own risk

? What do you wish to name the new devfile component mycomp
? What namespace do you want the devfile component to be created in default
 V  Validating devfile component [0ns]

Please use `odo push` command to create the component with source deployed
PS D:\ododev\sparse> ls                                                                                                 

    Directory: D:\ododev\sparse


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----       2020-05-06   1:52 PM                -async
d-----       2020-05-06   1:52 PM                -knative
d-----       2020-05-06   1:52 PM                -reactive
d-----       2020-05-06   1:52 PM                -reactive-crud
d-----       2020-05-06   1:52 PM                -testing
d-----       2020-05-06   1:52 PM                .mvn
d-----       2020-05-06   1:52 PM                .odo
d-----       2020-05-06   1:52 PM                .s2i
d-----       2020-05-06   1:52 PM                src
-a----       2020-05-06   1:52 PM             53 .dockerignore
-a----       2020-05-06   1:52 PM            295 .gitignore
-a----       2020-05-06   1:51 PM           2863 devfile.yaml
-a----       2020-05-06   1:52 PM          10078 mvnw
-a----       2020-05-06   1:52 PM           6609 mvnw.cmd
-a----       2020-05-06   1:52 PM           4964 pom.xml
-a----       2020-05-06   1:52 PM           2736 README.md
```


But when doing it manually via git, it only gives me the `getting-started/` folder as expected:
```
PS D:\ododev\quarkus-quickstarts> git sparse-checkout init --cone                                                       PS D:\ododev\quarkus-quickstarts> git sparse-checkout set getting-started                                               PS D:\ododev\quarkus-quickstarts> ls                                                                                    

    Directory: D:\ododev\quarkus-quickstarts


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----       2020-05-06   2:22 PM                getting-started
-a----       2020-05-06   2:21 PM            491 .gitignore
-a----       2020-05-06   2:21 PM           1808 azure-mvn-settings.xml
-a----       2020-05-06   2:21 PM            649 azure-pipelines.yml
-a----       2020-05-06   2:21 PM           2294 CONTRIBUTING.md
-a----       2020-05-06   2:21 PM           6148 gradlew
-a----       2020-05-06   2:21 PM           2942 gradlew.bat
-a----       2020-05-06   2:21 PM          11558 LICENSE
-a----       2020-05-06   2:21 PM          10379 mvnw
-a----       2020-05-06   2:21 PM           6789 mvnw.cmd
-a----       2020-05-06   2:21 PM           3315 pom.xml
-a----       2020-05-06   2:21 PM           6748 README.md
```

Reply:
So it seems it was the prefix that was causing this error, I have something in place where we add a path separator to the end of the pathToUnzip if there isn't one present and that seems to fix it in all aspects.

Reply:
... But I would be willing to hear your opinion on that change :)